nannyml/io/store/base.py

```diff
@@ -77,7 +77,7 @@ def load(self, as_type: Optional[type] = None, **load_args):
             self._logger.info(f'loading object from store "{self}"')
             obj = self._load(**load_args)
             if as_type and not (obj is None or isinstance(obj, as_type)):
```

Comment:
Actually, there is probably one more issue:
Why object is allowed to be None if as_type is specified?

Reply:
Good catch @smetam . I think the correct version should check if the object is `None` or not an instance of `as_type`. So:

```python
if as_type and (obj is None or not isinstance(obj, as_type)):
    raise StoreException(f'loaded object is not of type "{type}"')
```
I'll update this and add it to your PR.